public/jira/raw/rollback/rollback.ts

```diff
@@ -0,0 +1,311 @@
+/**
+ * This module is responsible for rolling back Jira issues to their previous states based on their changelogs.
+ * It processes changes to fields such as Sprint, Fix Versions, Parent associations, and Status,
+ * ensuring that issues are accurately reverted to their historical configurations.
+ */
+import { parseDateISOString } from "../../../date-helpers";
+import {
+  Change,
+  FixVersion,
+  JiraIssue,
+  Sprint,
+  Status,
+} from "../../shared/types";
+
+interface RolledBackMetadata {
+  rolledbackTo: Date;
+  didNotExist?: boolean;
+}
+
+interface RolledBackJiraIssue extends JiraIssue {
+  rollbackMetadata: RolledBackMetadata;
+}
+
+interface FieldLookup<T> {
+  ids: Map<string | number, T>;
+  names: Map<string, T>;
+}
+
+interface RollbackLookupData {
+  sprints: FieldLookup<Sprint>;
+  versions: FieldLookup<FixVersion>;
+  statuses: FieldLookup<Status>;
+}
+
+function getSprintNumbers(value: string): number[] | null {
+  if (value === "") {
+    return null;
+  } else {
+    return value.split(",").map((num) => +num);
+  }
+}
+
+function getSprintNames(value: string): string[] | null {
+  if (value === "") {
+    return null;
+  } else {
+    return value.split(",").map((name) => name.trim());
+  }
+}
+
+type FieldRollbackFunction = (
+  lastReturnValue: unknown,
```

Comment:
I kept this as unknown since the type of it differs based on the field function using it (Sprint, Fix Versions, etc..) 
Otherwise it will be a long type that lists all the possible types.
What do you think?

Reply:
Might wanna do the same for the object being returned and default it to any, otherwise when using the result of these function in a ts file you're going to need to validate each and every aspect of the value since `unknown` is very strict

Reply:
If you go this route, it'd be interesting to put the function type on the individual functions and see it there are any issues that arise typing them that way

Reply:
I could not explicitly put the function type on the functions since they are inline, but I tested it by pulling out the Sprint function in its own var and something like that worked fine

const sprintRollback: FieldRollbackFunction<Sprint, Sprint | Sprint [] | null> = (
  lastReturnValue: Sprint,
  change: Change,
  fieldName: string,
  { sprints }: RollbackLookupData
) => { ....}